jenkins-integration.sh

```diff
@@ -0,0 +1,162 @@
+curl --retry-all-errors --connect-timeout 5 --max-time 10 --retry 20 --retry-delay 0 --retry-max-time 120 'http://127.0.0.1:8091'
+
+# Set up CBS
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/nodes/self/controller/settings -d 'path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'index_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'cbas_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&' -d 'eventing_path=%2Fopt%2Fcouchbase%2Fvar%2Flib%2Fcouchbase%2Fdata&'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/node/controller/rename -d 'hostname=127.0.0.1'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/node/controller/setupServices -d 'services=kv%2Cn1ql%2Cindex'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/pools/default -d 'memoryQuota=3072' -d 'indexMemoryQuota=3072' -d 'ftsMemoryQuota=256'
+curl -u Administrator:password -v -X POST http://127.0.0.1:8091/settings/web -d 'password=password&username=Administrator&port=SAME'
+curl -u Administrator:password -v -X POST http://localhost:8091/settings/indexes -d indexerThreads=4 -d logLevel=verbose -d maxRollbackPoints=10 \
+-d storageMode=plasma -d memorySnapshotInterval=150 -d stableSnapshotInterval=40000
+
+sleep 10
+
+# Set up test environment variables for CBS runs
+export SG_TEST_USE_XATTRS=${XATTRS}
+export SG_TEST_USE_GSI=${GSI}
+export SG_TEST_COUCHBASE_SERVER_URL="${COUCHBASE_SERVER_PROTOCOL}://127.0.0.1"
+export SG_TEST_BACKING_STORE="Couchbase"
+export SG_TEST_BUCKET_POOL_SIZE=${SG_TEST_BUCKET_POOL_SIZE}
+export SG_TEST_TLS_SKIP_VERIFY=${TLS_SKIP_VERIFY}
+
+if [ "${SG_EDITION}" == "EE" ]; then
+    GO_TEST_FLAGS="${GO_TEST_FLAGS} -tags cb_sg_enterprise"
+fi
+
+go test ${GO_TEST_FLAGS} -coverprofile=coverage_int.out -coverpkg=github.com/couchbase/sync_gateway/${TARGET_PACKAGE} github.com/couchbase/sync_gateway/${TARGET_PACKAGE} 2>&1 | tee "${INT_LOG_FILE_NAME}.txt" | grep -E '(--- (FAIL|PASS|SKIP):|github.com/couchbase/sync_gateway(/.+)?\t|TEST: |panic: )'
+if [ "${PIPESTATUS[0]}" -ne "0" ]; then # If test exit code is not 0 (failed)
+  echo "Go test failed! Parsing logs to find cause..."
+  TEST_FAILED=true
+fi
+
+# Generate xunit test report that can be parsed by the JUnit Plugin
+~/go/bin/go2xunit -input "${INT_LOG_FILE_NAME}.txt" -output "${INT_LOG_FILE_NAME}.xml"
+
+# Get coverage
+~/go/bin/gocov convert "coverage_int.out" | ~/go/bin/gocov-xml > coverage_int.xml
+if [ "${RUN_WALRUS}" == "true" ]; then
+  ~/go/bin/gocov convert "coverage_walrus_ee.out" | ~/go/bin/gocov-xml > coverage_walrus_ee.xml
+  ~/go/bin/gocov convert "coverage_walrus_ce.out" | ~/go/bin/gocov-xml > coverage_walrus_ce.xml
+fi
+
+if [ "${TEST_FAILED}" = true ] ; then
+  # If output contained `FAIL:`
+  if grep -q 'FAIL:' "${INT_LOG_FILE_NAME}.txt"; then
+    # Test failure, so mark as unstable
+    exit 50
+  else
+    # Test setup failure or unrecovered panic, so mark as failed
+    exit 1
+  fi
+fi
```

Comment:
Nitpick: eof newline

```suggestion
fi

```